pkg/bigquery/db.go

```diff
@@ -1303,3 +1303,76 @@ func (d *Client) GetDatabaseSummary(ctx context.Context) (*ansisql.DBDatabase, e
 
 	return summary, nil
 }
+
+func (d *Client) GetDatabaseSummaryForSchemas(ctx context.Context, schemas []string) (*ansisql.DBDatabase, error) {
+	if err := d.ensureClientInitialized(ctx); err != nil {
+		return nil, err
+	}
+
+	projectID := d.config.ProjectID
+
+	summary := &ansisql.DBDatabase{
+		Name:    projectID,
+		Schemas: []*ansisql.DBSchema{},
+	}
+
+	mu := sync.Mutex{}
+	var errs []error
+
+	workers := max(runtime.NumCPU(), 8)
+	p := pool.New().WithMaxGoroutines(workers)
+
+	// Only iterate over requested schemas (datasets)
+	for _, schemaName := range schemas {
+		p.Go(func() {
```

Comment:
create a variable in the loop `schemaName := schemaName`  to avoid goroutines capturing the shared range variable. otherwise you will have multiple goroutines working on the same schema